src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java

```diff
@@ -416,4 +325,41 @@ public void triggerDeathAchievement(final DamageSource source, final EntityCitiz
     {
 
     }
+
+    /**
+     * Method called when a stack is pickup by the entity.
+     *
+     * @param pickedUpStack The stack that is being picked up.
+     * @return true when the stack has been used to resolve a request, false when not.
+     */
+    public boolean onStackPickUp(@NotNull final ItemStack pickedUpStack)
+    {
+        if (getCitizen().getWorkBuilding() != null)
+        {
+            if (getCitizen().getWorkBuilding().overruleNextOpenRequestOfCitizenWithStack(getCitizen(), pickedUpStack.copy()))
+            {
+                return true;
+            }
+        }
+
+        if (getCitizen().getHomeBuilding() != null)
+        {
+            if (getCitizen().getHomeBuilding().overruleNextOpenRequestOfCitizenWithStack(getCitizen(), pickedUpStack.copy()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)